documents/development/README.md

```diff
@@ -0,0 +1,537 @@
+install(DIRECTORY include/
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/celix_my_lib)
+install(DIRECTORY ${CMAKE_BINARY_DIR}/celix/gen/includes/my_lib/
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/celix_my_lib)
+```
+
+### Configuring Symbol Visibility for C/C++ Bundles
+
+For bundle, symbol visibility will default be configured to hidden. This can be default by providing
+the `DO_NOT_CONFIGURE_SYMBOL_VISIBILITY` option to the CMake `add_celix_bundle` function.
+
+If symbol visibility is not configured in the `add_celix_bundle`, symbol visibility should be configured the same
+way as a shared library.
+
+```cmake
+add_celix_bundle(my_bundle
+    SOURCES src/my_bundle.c
+    SYMBOLIC_NAME "apache_celix_my_bundle"
+    NAME "Apache Celix My Bundle"
+    FILENAME "celix_my_bundle"
+    VERSION "1.0.0"
+    GROUP "celix/my_bundle_group"
+)
+add_library(celix::my_bundle ALIAS my_bundle)
+```
+
+## Branch naming
+
+- Prefix feature branches with `feature/`, hotfix branches with `hotfix/`, bugfix branches with `bugfix/` 
+  and release branches with `release/`.
+- If you are working on an issue, prefix the branch name with the issue number. E.g., `feature/1234-add-feature`.
+- Hotfix branches are for urgent fixes that need to be applied as soon as possible.
+- Use short and descriptive branch names.
+
+## Commit Messages
+    
+- Utilize the imperative mood when writing commit messages (e.g., "Add feature" instead of "Adds feature" 
+  or "Added feature"). This style aligns with git's auto-generated messages for merge commits or revert actions.
+- Ensure that commit messages are descriptive and provide meaningful context. 
+- Keep the first line of the commit message concise, ideally under 50 characters. 
+  This summary line serves as a quick overview of the change and should be easy to read in git logs. 
+- If more context is needed, separate the summary line from the body with a blank line. 
+  The body can provide additional details, explanations, or reasoning behind the changes. 
+  Aim to keep each line of the commit message body wrapped at around 72 characters for optimal readability.
+- Use bullet points, numbered lists, or other formatting conventions when listing multiple changes or points in the 
+  commit message body to improve readability.
+- When applicable, reference related issues, bug reports, or pull requests in the commit message body to 
+  provide additional context and establish connections between the commit and the larger project.
+  - If your commit fixes, closes, or resolves an issue, use one of these keywords followed by the issue number 
+    (e.g., "fixes #42", "closes #42", or "resolves #42"). 
+  - If you want to reference an issue without closing it, simply mention the issue number 
+    (e.g., "related to #42" or "#42").
+
+## Benchmarking
+
+- When needed, use benchmarking to measure performance.
+- Use the Google Benchmark framework for benchmarking.
```

Comment:
Why only aim at using Google Benchmark? If someone finds a better tool to do the job, that's up to them IMO (up to the user).

Reply:
To keep the external libs a bit more manageable. If there is a good reason to use a different benchmark lib, I see no problem with updating the guidlines.